### PR TITLE
[cxx Interop] Disambiguate methods within the same class with same name but differing "constness"

### PR DESCRIPTION
**Pull request by Huddie**
When importing CXX methods into Swift the compiler had issues with classes (Records) that contained 2+ methods that differed in "constness" but had the same name

Example:
```cpp
int increment(int a);
int increment(int a) const;
```

This PR appends "Mutating" to the end of the mutable function to disambiguate when importing into swift.

the result of the above would then be:

```swift
func incrementMutating(_ a: Int32) -> Int32
func increment(_ a: Int32) -> Int32
```

Determining when 2 methods conflict is a seemingly hard problem. Name alone is not enough to get it right 100% of the time since you could have a class with:

```cpp
int increment(int a);
int increment(int a, int b) const;
```

That said, such patterns seem less common and so this PR uses name alone to disambiguate, recognizing that it may result in some confusing outputs in more uncommon circumstances

The above example would result in:

```swift
func incrementMutating(_ a: Int32) -> Int32
func increment(_ a: Int32, _ b: Int32) -> Int32
```

even though the 2 methods do NOT conflict.

### CODE REVIEW

**Review by zoecarver — APPROVED** [review]
This PR is fantastic. Thank you, Ehud!

**Comment by zoecarver** [comment]
@swift-ci please smoke test.

**Comment by zoecarver** [comment]
@swift-ci please smoke test.

File: test/Interop/Cxx/class/method/Inputs/ambiguous_methods.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    a += b;
+  }
+
+  void increment(int &a, int b) const {
+    a += b;
+  }
+
+  // No input with output
+  int numberOfMutableMethodsCalled() { return ++mutableMethodsCalledCount; }
+  int numberOfMutableMethodsCalled() const { return mutableMethodsCalledCount; }
```
Comment (hyp) [inline test/Interop/Cxx/class/method/Inputs/ambiguous_methods.h:42]: It looks like your test cases are in a pattern of:
```
-  mutating method
- const method
```

Could you also add another method pair in reverse order, i.e. 
```
- const method
- mutating method
```

**Comment by hyp** [comment]
LG with one suggestion.

However, I believe we need a discussion if adding 'Mutating' at the end of the method is the right approach with respect to the naming convention. A 'mutating...' prefix might be ergonomic. Could you write up a short pitch for this on the Swift forums where we can have a discussion about the naming?

File: test/Interop/Cxx/class/method/Inputs/ambiguous_methods.h (hunk 1 above)
Comment (Huddie) [reply test/Interop/Cxx/class/method/Inputs/ambiguous_methods.h:42]: Great idea!

**Pull request merged**